nilearn/decomposition/_base.py

```diff
@@ -497,15 +505,29 @@ def fit(
         )
         self._raw_fit(data)
 
-        # Create and fit NiftiMapsMasker for transform
+        # Create and fit appropriate MapsMasker for transform
         # and inverse_transform
-        self.nifti_maps_masker_ = NiftiMapsMasker(
-            self.components_img_,
-            self.masker_.mask_img_,
-            resampling_target="maps",
-        )
-
-        self.nifti_maps_masker_.fit()
+        if isinstance(self.masker_, SurfaceMasker):
+            self.maps_masker_ = SurfaceMapsMasker(
+                self.components_img_, self.masker_.mask_img_
+            )
+            self.surface_maps_masker_.fit()
+        else:
+            self.maps_masker_ = NiftiMapsMasker(
+                self.components_img_,
+                self.masker_.mask_img_,
+                resampling_target="maps",
+            )
+            # TODO: remove in 0.11.4
+            self.nifti_maps_masker_ = self.maps_masker_
+            warnings.warn(
+                FutureWarning,
+                message="The nifti_maps_masker_ attribute is deprecated and"
+                "will be removed in Nilearn 0.11.3. Please use "
+                "maps_masker_ instead.",
+                stacklevel=2,
+            )
```

Comment:
maybe mode this into a property
```python
            @property
            def nifti_maps_masker_(self):
              warnings.warn(
                FutureWarning,
                message="The nifti_maps_masker_ attribute is deprecated and"
                "will be removed in Nilearn 0.11.3. Please use "
                "maps_masker_ instead.",
                stacklevel=2,
              )
              return self.maps_masker_
```

Reply:
Doing this refactoring made me realize that I didn't update _BaseDecomposition's transform and inverse_transform to work with surface masker. So now I am using `self.maps_masker_` in those too (and hence avoiding doing if else). That wouldn't work with the property like this then